go/kbfs/libkbfs/subscription_manager.go

```diff
@@ -199,9 +216,60 @@ func (sm *subscriptionManager) unregisterForChangesLocked(fb data.FolderBranch)
 	sm.subscriptionCountByFolderBranch[fb]--
 }
 
-func (sm *subscriptionManager) subscribePath(ctx context.Context,
+func (sm *subscriptionManager) makePathSubscriptionDebouncedNotify(
+	ref pathSubscriptionRef, limit rate.Limit) *debouncedNotify {
+	return debounce(func() {
+		sm.lock.RLock()
```

Comment:
I know it's annoying, but it would make me feel better to package most of this up in a separate function that can defer the unlock, and just return the needed data.

Reply:
Whoops I forgot to do that (you mentioned in your last review). Will do thanks!